src/cartridges/bm_adyen/cartridge/static/default/adyenSettings.js

```diff
@@ -3,84 +3,174 @@ document.addEventListener('DOMContentLoaded', () => {
   const cancelButton = document.querySelector('#settingsFormCancelButton');
   const formButtons = Array.from(document.getElementsByClassName('formButton'));
+  const testConnectionButton = document.querySelector('#testConnectionButton');
+  const togglePassword = document.querySelector("#togglePassword");
+  const toggleApi = document.querySelector("#toggleApi");
+  const formBody = document.querySelector('#formBody');
+  const password = document.querySelector("#notificationsPassword");
+  const merchAccount = document.getElementById("merchantAccount");
+  const apiKeyVal = document.getElementById('apiKey');
   const changedSettings = [];
+  testConnectionButton.disabled = true;
 
   function settingChanged(key, value) {
-    const settingIndex = changedSettings.findIndex(
-      (setting) => setting.key === key,
-    );
+    const settingIndex = changedSettings.findIndex(setting => {
+      return setting.key === key;
+    });
 
     if (settingIndex >= 0) {
-      changedSettings[settingIndex] = { key, value };
+      changedSettings[settingIndex] = {
+        key,
+        value
+      };
     } else {
-      changedSettings.push({ key, value });
+      changedSettings.push({
+        key,
+        value
+      });
     }
   }
 
   function enableformButtons() {
-    formButtons.forEach((button) => {
+    for (const button of formButtons) {
       button.classList.remove('disabled');
       button.classList.add('enabled');
       form.removeEventListener('input', enableformButtons);
-    });
+    }
+
+    ;
   }
 
   function disableFormButtons() {
-    formButtons.forEach((button) => {
+    for (const button of formButtons) {
       button.classList.remove('enabled');
       button.classList.add('disabled');
       form.removeEventListener('input', enableformButtons);
-    });
+    }
+  } // add event for save button availability on form change.
+
+
+  function clickAndHide() {
+    document.getElementById('settingsFormSubmitButton').click();
+    document.getElementById('saveChangesAlert').hide();
+    document.getElementById('notSavedChangesAlert').hide();
+    document.getElementById('testConnectionButton').disabled = true;
+  }
+
+  function showAlerts() {
+    document.getElementById('saveChangesAlert').show();
+    document.getElementById('notSavedChangesAlert').show();
   }
 
-  // add event for save button availability on form change.
-  form.addEventListener('input', enableformButtons);
+  function renderSafari(){
+    if (/^((?!chrome|android).)*safari/i.test(navigator.userAgent)){
+      formBody.style.setProperty('padding-top', '3rem');
+    }
+  }
+
+  testConnectionButton.addEventListener('click', clickAndHide); // add event listener to hide the alerts in case of test connection
+
+  form.addEventListener('input', enableformButtons); // add event listener to maintain form updates
+
+  submitButton.addEventListener('click', showAlerts); // add event listener to show alerts just in case of save button clicked
```

Comment:
let's think of a way to make the function names more descriptive that you wouldn't need to add comments anymore. For example this function name could be `showAlertsOnSave`? 
and the function `clickAndHide` could possibly be `hideAlertsOnTest`?